### PR TITLE
AGW-2221: Remove Application and ApiKey from TITA API

### PR DESCRIPTION
**Pull request by mtqp**

### CODE REVIEW

File: integration-tests/src/test/java/com/mulesoft/anypoint/templates/HeaderInjectionPolicyTestCase.java [hunk 1]
```diff
@@ -22,75 +17,88 @@
...
-    return Tita.engine().runtime().standalone()
-        .withApplications(buildTestApplication(APP, "mule-config-test-flows.xml"));
-  }
-
-  public HeaderInjectionPolicyTestCase(RuntimeBuilder runtimeBuilder,
-                                       PolicySupplier policySupplier) {
-    super(runtimeBuilder, policySupplier);
+  @Before
+  public void setUp() {
+    runtime().deploy(applicationBuilder.build());
```
Comment (balbifm) [inline integration-tests/src/test/java/com/mulesoft/anypoint/templates/HeaderInjectionPolicyTestCase.java]: Try to deploy app during runtime building

File: integration-tests/src/test/java/com/mulesoft/anypoint/templates/HeaderInjectionPolicyTestCase.java [hunk 2]
```diff
@@ -22,75 +17,88 @@
...
+                                       ApplicationBuilder applicationBuilder) {
+    super(runtimeBuilder, policySupplier);
+    this.api = api;
+    this.apiAddingHeader = apiAddingHeader;
+    this.applicationBuilder = applicationBuilder;
+  }
 
   @Parameterized.Parameters
   public static Collection<Object[]> testConfigurationData() {
+    List<ApiKey> apiKeys = Tita.engine().api().create(2);
```
Comment (balbifm) [inline integration-tests/src/test/java/com/mulesoft/anypoint/templates/HeaderInjectionPolicyTestCase.java]: Create JIRA for ER

Comment (mtqp) [reply integration-tests/src/test/java/com/mulesoft/anypoint/templates/HeaderInjectionPolicyTestCase.java]: AGW-2243: Read application to automatically determine autodiscoveries and listener config

File: integration-tests/src/test/java/com/mulesoft/anypoint/templates/HeaderInjectionPolicyTestCase.java (hunk 1 above)
Comment (mtqp) [reply integration-tests/src/test/java/com/mulesoft/anypoint/templates/HeaderInjectionPolicyTestCase.java]: Tried it but somehow it doesn't work... There was a reason why, I just don't remember it! Create: AGW-2244 to review it.

**Pull request merged**